File: sample/src/main/java/me/xiaopan/sketchsample/bean/UnsplashImage.java
package me.xiaopan.sketchsample.bean;

import android.text.TextUtils;

import com.google.gson.annotations.SerializedName;

public class UnsplashImage {
    @SerializedName("id")
    public String id;

    @SerializedName("created_at")
    public String createdAt;

    @SerializedName("updated_at")
    public String updatedAt;

    @SerializedName("width")
    public int width;

    @SerializedName("height")
    public int height;

    @SerializedName("color")
    public String color;

    @SerializedName("user")
    public User user;

    @SerializedName("urls")
    public Urls urls;

    @SerializedName("likes")
    public int likes;

    @SerializedName("liked_by_user")
    public boolean likedByUser;

    private transient String formattedUpdateDate;

    public String getFormattedUpdateDate() {
        if (formattedUpdateDate == null && !TextUtils.isEmpty(updatedAt) && updatedAt.length() >= 10) {
            formattedUpdateDate = updatedAt.substring(0, 10).replace("-", ".");
        }
        return formattedUpdateDate;
    }

    public static class User {
        @SerializedName("id")
        public String id;

        @SerializedName("updatedAt")
        public String updated_at;

        @SerializedName("username")
        public String username;

        @SerializedName("name")
        public String name;

        @SerializedName("first_name")
        public String firstName;

        @SerializedName("last_name")
        public String lastName;

        @SerializedName("portfolio_url")
        public String portfolioUrl;

        @SerializedName("bio")
        public String bio;

        @SerializedName("location")
        public String location;

        @SerializedName("total_likes")
        public int totalLikes;

        @SerializedName("total_photos")
        public int totalPhotos;

        @SerializedName("total_collections")
        public int totalCollections;

        @SerializedName("profile_image")
        public ProfileImage profileImage;

        @SerializedName("links")
        public Links links;

        public static class ProfileImage {
            @SerializedName("small")
            public String small;

            @SerializedName("medium")
            public String medium;

            @SerializedName("large")
            public String large;
        }

        public static class Links {
            @SerializedName("self")
            public String self;

            @SerializedName("html")
            public String html;

            @SerializedName("photos")
            public String photos;

            @SerializedName("likes")
            public String likes;

            @SerializedName("portfolio")
            public String portfolio;

            @SerializedName("following")
            public String following;

            @SerializedName("followers")
            public String followers;
        }
    }

    public static class Urls {
        @SerializedName("raw")
        public String raw;

        @SerializedName("full")
        public String full;

        @SerializedName("regular")
        public String regular;

        @SerializedName("small")
        public String small;

        @SerializedName("thumb")
        public String thumb;
    }
}

File: sample/src/main/java/me/panpf/sketch/sample/bean/UnsplashImage.kt
package me.panpf.sketch.sample.bean

import android.text.TextUtils

import com.google.gson.annotations.SerializedName

class UnsplashImage {
    @SerializedName("id")
    var id: String? = null

    @SerializedName("created_at")
    var createdAt: String? = null

    @SerializedName("updated_at")
    var updatedAt: String? = null

    @SerializedName("width")
    var width: Int = 0

    @SerializedName("height")
    var height: Int = 0

    @SerializedName("color")
    var color: String? = null

    @SerializedName("user")
    var user: User? = null

    @SerializedName("urls")
    var urls: Urls? = null

    @SerializedName("likes")
    var likes: Int = 0

    @SerializedName("liked_by_user")
    var likedByUser: Boolean = false

    @Transient private var formattedUpdateDate: String? = null

    fun getFormattedUpdateDate(): String {
        if (formattedUpdateDate == null && !TextUtils.isEmpty(updatedAt) && updatedAt!!.length >= 10) {
            formattedUpdateDate = updatedAt!!.substring(0, 10).replace("-", ".")
        }
        return formattedUpdateDate!!
    }

    class User {
        @SerializedName("id")
        var id: String? = null

        @SerializedName("updatedAt")
        var updated_at: String? = null

        @SerializedName("username")
        var username: String? = null

        @SerializedName("name")
        var name: String? = null

        @SerializedName("first_name")
        var firstName: String? = null

        @SerializedName("last_name")
        var lastName: String? = null

        @SerializedName("portfolio_url")
        var portfolioUrl: String? = null

        @SerializedName("bio")
        var bio: String? = null

        @SerializedName("location")
        var location: String? = null

        @SerializedName("total_likes")
        var totalLikes: Int = 0

        @SerializedName("total_photos")
        var totalPhotos: Int = 0

        @SerializedName("total_collections")
        var totalCollections: Int = 0

        @SerializedName("profile_image")
        var profileImage: ProfileImage? = null

        @SerializedName("links")
        var links: Links? = null

        class ProfileImage {
            @SerializedName("small")
            var small: String? = null

            @SerializedName("medium")
            var medium: String? = null

            @SerializedName("large")
            var large: String? = null
        }

        class Links {
            @SerializedName("self")
            var self: String? = null

            @SerializedName("html")
            var html: String? = null

            @SerializedName("photos")
            var photos: String? = null

            @SerializedName("likes")
            var likes: String? = null

            @SerializedName("portfolio")
            var portfolio: String? = null

            @SerializedName("following")
            var following: String? = null

            @SerializedName("followers")
            var followers: String? = null
        }
    }

    class Urls {
        @SerializedName("raw")
        var raw: String? = null

        @SerializedName("full")
        var full: String? = null

        @SerializedName("regular")
        var regular: String? = null

        @SerializedName("small")
        var small: String? = null

        @SerializedName("thumb")
        var thumb: String? = null
    }
}

